Library/Homebrew/extend/os/mac/system_config.rb

```diff
@@ -48,9 +48,14 @@ def core_tap_config(out = $stdout)
         sig { returns(T.nilable(String)) }
         def metal_toolchain
           @metal_toolchain ||= T.let(if MacOS::Xcode.installed? || MacOS::CLT.installed?
-                                       result = SystemCommand.run("xcrun", args: ["metal", "-v"],
-                                       print_stderr: false, print_stdout: false)
-                                       "present" if result.success?
+                                       result = SystemCommand.run("xcrun", args: ["--find", "metal"],
+                                                                  print_stderr: false, print_stdout: false)
+                                       # MetalToolchain-v<major>.<letter_num>.<build>.<minor>
+                                       if result.success? &&
+                                          result.stdout =~ /MetalToolchain-v(\d+)\.(\d+)\.(\d+)\.(\d+)/
```

Comment:
Using named groups might help a bit with readability here. Also using `.match` which will return `MatchData` on success so can avoid the `T.must`.